### PR TITLE
Remove scotch/metis link

### PR DESCRIPTION
**Pull request by minrk**
#111 fixed linking because scotch was linked but not expressed as a dependency, resulting in dll load errors when there's a scotch update.

But I think this is the 'real' fix, because mumps dependencies shouldn't be linked transitively (that's the case for static mumps, not the shared libraries we have in conda-forge).

Let's wait to merge until we can check the DSO inspection output from conda-build to make sure a link isn't still there that's not expressed in the dependencies.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -26,11 +26,10 @@ requirements:
     - cmake  # [win]
     - make  # [not win]
     - wget  # [not win]
-    - pkgconfig
+    - pkg-config
```
Comment (minrk) [inline recipe/meta.yaml:29]: btw, `pkgconfig` is a Python package, `pkg-config` is command-line tool

**Comment by traversaro** [comment]
analysis seems good:

~~~
2024-09-26T07:50:24.6247970Z    INFO (ipopt,bin/ipopt): lib/libipoptamplinterface.so.3 found in this package
2024-09-26T07:50:24.6261057Z    INFO (ipopt,bin/ipopt): lib/libipopt.so.3 found in this package
2024-09-26T07:50:24.6274041Z    INFO (ipopt,bin/ipopt): Needed DSO lib/libstdc++.so.6 found in conda-forge/linux-64::libstdcxx==14.1.0=hc0a3c3a_1
2024-09-26T07:50:24.6285250Z    INFO (ipopt,bin/ipopt): Needed DSO lib/libgcc_s.so.1 found in conda-forge/linux-64::libgcc==14.1.0=h77fa898_1
2024-09-26T07:50:24.6304173Z    INFO (ipopt,bin/ipopt): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
2024-09-26T07:50:24.6369084Z    INFO (ipopt,bin/ipopt_sens): lib/libsipopt.so.3 found in this package
2024-09-26T07:50:24.6382099Z    INFO (ipopt,bin/ipopt_sens): lib/libipoptamplinterface.so.3 found in this package
2024-09-26T07:50:24.6394695Z    INFO (ipopt,bin/ipopt_sens): lib/libipopt.so.3 found in this package
2024-09-26T07:50:24.6405985Z    INFO (ipopt,bin/ipopt_sens): Needed DSO lib/libasl.so.3 found in conda-forge/linux-64::ampl-mp==3.1.0=h2cc385e_1006
2024-09-26T07:50:24.6418778Z    INFO (ipopt,bin/ipopt_sens): Needed DSO lib/libstdc++.so.6 found in conda-forge/linux-64::libstdcxx==14.1.0=hc0a3c3a_1
2024-09-26T07:50:24.6430558Z    INFO (ipopt,bin/ipopt_sens): Needed DSO lib/libgcc_s.so.1 found in conda-forge/linux-64::libgcc==14.1.0=h77fa898_1
2024-09-26T07:50:24.6446651Z    INFO (ipopt,bin/ipopt_sens): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
2024-09-26T07:50:24.6614725Z    INFO (ipopt,lib/libipoptamplinterface.so.3.14.16): lib/libipopt.so.3 found in this package
2024-09-26T07:50:24.6631731Z    INFO (ipopt,lib/libipoptamplinterface.so.3.14.16): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/librt.so.1 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
2024-09-26T07:50:24.6641568Z    INFO (ipopt,lib/libipoptamplinterface.so.3.14.16): Needed DSO lib/liblapack.so.3 found in conda-forge/linux-64::liblapack==3.9.0=7_ha36c22a_netlib
2024-09-26T07:50:24.6652902Z    INFO (ipopt,lib/libipoptamplinterface.so.3.14.16): Needed DSO lib/libblas.so.3 found in conda-forge/linux-64::libblas==3.9.0=7_ha36c22a_netlib
2024-09-26T07:50:24.6666310Z    INFO (ipopt,lib/libipoptamplinterface.so.3.14.16): Needed DSO lib/libspral.so found in conda-forge/linux-64::libspral==2024.05.08=h831f25b_3
2024-09-26T07:50:24.6681871Z    INFO (ipopt,lib/libipoptamplinterface.so.3.14.16): Needed DSO lib/libdmumps_seq.so found in conda-forge/linux-64::mumps-seq==5.7.3=h96124f5_4
2024-09-26T07:50:24.6696211Z    INFO (ipopt,lib/libipoptamplinterface.so.3.14.16): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libdl.so.2 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
2024-09-26T07:50:24.6706106Z    INFO (ipopt,lib/libipoptamplinterface.so.3.14.16): Needed DSO lib/libasl.so.3 found in conda-forge/linux-64::ampl-mp==3.1.0=h2cc385e_1006
2024-09-26T07:50:24.6716900Z    INFO (ipopt,lib/libipoptamplinterface.so.3.14.16): Needed DSO lib/libstdc++.so.6 found in conda-forge/linux-64::libstdcxx==14.1.0=hc0a3c3a_1
2024-09-26T07:50:24.6731256Z    INFO (ipopt,lib/libipoptamplinterface.so.3.14.16): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libm.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
2024-09-26T07:50:24.6746904Z    INFO (ipopt,lib/libipoptamplinterface.so.3.14.16): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
2024-09-26T07:50:24.6756438Z    INFO (ipopt,lib/libipoptamplinterface.so.3.14.16): Needed DSO lib/libgcc_s.so.1 found in conda-forge/linux-64::libgcc==14.1.0=h77fa898_1
2024-09-26T07:50:24.6939857Z    INFO (ipopt,lib/libsipopt.so.3.14.16): lib/libipopt.so.3 found in this package
2024-09-26T07:50:24.6954927Z    INFO (ipopt,lib/libsipopt.so.3.14.16): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/librt.so.1 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
2024-09-26T07:50:24.6966572Z    INFO (ipopt,lib/libsipopt.so.3.14.16): Needed DSO lib/liblapack.so.3 found in conda-forge/linux-64::liblapack==3.9.0=7_ha36c22a_netlib
2024-09-26T07:50:24.6977223Z    INFO (ipopt,lib/libsipopt.so.3.14.16): Needed DSO lib/libblas.so.3 found in conda-forge/linux-64::libblas==3.9.0=7_ha36c22a_netlib
2024-09-26T07:50:24.6988614Z    INFO (ipopt,lib/libsipopt.so.3.14.16): Needed DSO lib/libspral.so found in conda-forge/linux-64::libspral==2024.05.08=h831f25b_3
2024-09-26T07:50:24.6998194Z    INFO (ipopt,lib/libsipopt.so.3.14.16): Needed DSO lib/libdmumps_seq.so found in conda-forge/linux-64::mumps-seq==5.7.3=h96124f5_4
2024-09-26T07:50:24.7014145Z    INFO (ipopt,lib/libsipopt.so.3.14.16): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libdl.so.2 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
2024-09-26T07:50:24.7027849Z    INFO (ipopt,lib/libsipopt.so.3.14.16): Needed DSO lib/libstdc++.so.6 found in conda-forge/linux-64::libstdcxx==14.1.0=hc0a3c3a_1
2024-09-26T07:50:24.7046177Z    INFO (ipopt,lib/libsipopt.so.3.14.16): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libm.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
2024-09-26T07:50:24.7063688Z    INFO (ipopt,lib/libsipopt.so.3.14.16): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
2024-09-26T07:50:24.7098667Z    INFO (ipopt,lib/libsipopt.so.3.14.16): Needed DSO lib/libgcc_s.so.1 found in conda-forge/linux-64::libgcc==14.1.0=h77fa898_1
2024-09-26T07:50:24.7792000Z    INFO (ipopt,lib/libipopt.so.3.14.16): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/librt.so.1 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
2024-09-26T07:50:24.7802921Z    INFO (ipopt,lib/libipopt.so.3.14.16): Needed DSO lib/liblapack.so.3 found in conda-forge/linux-64::liblapack==3.9.0=7_ha36c22a_netlib
2024-09-26T07:50:24.7811012Z    INFO (ipopt,lib/libipopt.so.3.14.16): Needed DSO lib/libblas.so.3 found in conda-forge/linux-64::libblas==3.9.0=7_ha36c22a_netlib
2024-09-26T07:50:24.7820564Z    INFO (ipopt,lib/libipopt.so.3.14.16): Needed DSO lib/libspral.so found in conda-forge/linux-64::libspral==2024.05.08=h831f25b_3
2024-09-26T07:50:24.7831828Z    INFO (ipopt,lib/libipopt.so.3.14.16): Needed DSO lib/libdmumps_seq.so found in conda-forge/linux-64::mumps-seq==5.7.3=h96124f5_4
2024-09-26T07:50:24.7846278Z    INFO (ipopt,lib/libipopt.so.3.14.16): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libdl.so.2 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
2024-09-26T07:50:24.7856434Z    INFO (ipopt,lib/libipopt.so.3.14.16): Needed DSO lib/libstdc++.so.6 found in conda-forge/linux-64::libstdcxx==14.1.0=hc0a3c3a_1
2024-09-26T07:50:24.7872223Z    INFO (ipopt,lib/libipopt.so.3.14.16): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libm.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
2024-09-26T07:50:24.7888187Z    INFO (ipopt,lib/libipopt.so.3.14.16): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
2024-09-26T07:50:24.7899711Z    INFO (ipopt,lib/libipopt.so.3.14.16): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/ld-linux-x86-64.so.2 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
2024-09-26T07:50:24.7909511Z    INFO (ipopt,lib/libipopt.so.3.14.16): Needed DSO lib/libgcc_s.so.1 found in conda-forge/linux-64::libgcc==14.1.0=h77fa898_1
2024-09-26T07:50:24.7933949Z WARNING (ipopt): dso library package conda-forge/linux-64::libgfortran5==14.1.0=hc5f4f2c_1 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
2024-09-26T07:50:24.7937545Z WARNING (ipopt): run-exports library package conda-forge/linux-64::scotch==7.0.4=hd53efc5_5 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
~~~

File: recipe/meta.yaml [hunk 2]
```diff
@@ -26,11 +26,10 @@ requirements:
...
   host:
     - libblas
     - liblapack
-    - metis
     - mumps-seq
     - scotch
```
Comment (traversaro) [inline recipe/meta.yaml:34]:
```suggestion
```

Comment (minrk) [reply recipe/meta.yaml]: oops, that's what I get for rebasing concurrent PRs. Thanks!

**Comment by traversaro** [comment]
@conda-forge-admin please rerender

**Comment by traversaro** [comment]
New output:

~~~
   INFO: sysroot: '/home/conda/feedstock_root/build_artifacts/ipopt_1727342737576/_build_env/x86_64-conda-linux-gnu/sysroot/' files: '['usr/share/zoneinfo', 'usr/share/locale/zh_TW/LC_MESSAGES/libc.mo', 'usr/share/locale/zh_CN/LC_MESSAGES/libc.mo', 'usr/share/locale/vi/LC_MESSAGES/libc.mo']'
   INFO (ipopt,bin/ipopt): lib/libipoptamplinterface.so.3 found in this package
   INFO (ipopt,bin/ipopt): lib/libipopt.so.3 found in this package
   INFO (ipopt,bin/ipopt): Needed DSO lib/libstdc++.so.6 found in conda-forge/linux-64::libstdcxx==14.1.0=hc0a3c3a_1
   INFO (ipopt,bin/ipopt): Needed DSO lib/libgcc_s.so.1 found in conda-forge/linux-64::libgcc==14.1.0=h77fa898_1
   INFO (ipopt,bin/ipopt): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
   INFO (ipopt,lib/libipopt.so.3.14.16): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/librt.so.1 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
   INFO (ipopt,lib/libipopt.so.3.14.16): Needed DSO lib/liblapack.so.3 found in conda-forge/linux-64::liblapack==3.9.0=7_ha36c22a_netlib
   INFO (ipopt,lib/libipopt.so.3.14.16): Needed DSO lib/libblas.so.3 found in conda-forge/linux-64::libblas==3.9.0=7_ha36c22a_netlib
   INFO (ipopt,lib/libipopt.so.3.14.16): Needed DSO lib/libspral.so found in conda-forge/linux-64::libspral==2024.05.08=h831f25b_3
   INFO (ipopt,lib/libipopt.so.3.14.16): Needed DSO lib/libdmumps_seq.so found in conda-forge/linux-64::mumps-seq==5.7.3=hcd84eb1_5
   INFO (ipopt,lib/libipopt.so.3.14.16): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libdl.so.2 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
   INFO (ipopt,lib/libipopt.so.3.14.16): Needed DSO lib/libstdc++.so.6 found in conda-forge/linux-64::libstdcxx==14.1.0=hc0a3c3a_1
   INFO (ipopt,lib/libipopt.so.3.14.16): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libm.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
   INFO (ipopt,lib/libipopt.so.3.14.16): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
   INFO (ipopt,lib/libipopt.so.3.14.16): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/ld-linux-x86-64.so.2 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
   INFO (ipopt,lib/libipopt.so.3.14.16): Needed DSO lib/libgcc_s.so.1 found in conda-forge/linux-64::libgcc==14.1.0=h77fa898_1
   INFO (ipopt,lib/libipoptamplinterface.so.3.14.16): lib/libipopt.so.3 found in this package
   INFO (ipopt,lib/libipoptamplinterface.so.3.14.16): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/librt.so.1 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
   INFO (ipopt,lib/libipoptamplinterface.so.3.14.16): Needed DSO lib/liblapack.so.3 found in conda-forge/linux-64::liblapack==3.9.0=7_ha36c22a_netlib
   INFO (ipopt,lib/libipoptamplinterface.so.3.14.16): Needed DSO lib/libblas.so.3 found in conda-forge/linux-64::libblas==3.9.0=7_ha36c22a_netlib
   INFO (ipopt,lib/libipoptamplinterface.so.3.14.16): Needed DSO lib/libspral.so found in conda-forge/linux-64::libspral==2024.05.08=h831f25b_3
   INFO (ipopt,lib/libsipopt.so.3.14.16): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 found in CDT/compiler package conda-forge/noarch::sysroot_linux-64==2.17=h4a8ded7_16
   INFO (ipopt,lib/libsipopt.so.3.14.16): Needed DSO lib/libgcc_s.so.1 found in conda-forge/linux-64::libgcc==14.1.0=h77fa898_1
WARNING (ipopt): dso library package conda-forge/linux-64::libgfortran5==14.1.0=hc5f4f2c_1 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
~~~

I think this is a good to go, what do you think @minrk ?

**Review by traversaro — APPROVED** [review]

**Comment by minrk** [comment]
Yup, looks right to me!

**Comment by minrk** [comment]
For future reference and for the repodata patch, linking scotch without expressing the dependency started in 3.13.3 build 2 with #50 . I will probably only do the repodata patch for 3.14.16 because finding the scotch version used at build time will be a pain. Since mumps is pinned x.y.z transitively pins libscotch also to x.y.z, it's really only builds against a given version of mumps when that version of mumps is later rebuilt against newer scotch that are affected, which I don't think happens that often, and likely means the latest build of most ipopt versions is fine and can't break in the future because past versions of mumps aren't getting fresh builds.

**Comment by traversaro** [comment]
I am afraid there was a regression: https://github.com/conda-forge/ipopt-feedstock/issues/114 .

**Pull request merged**